webpack/scenes/ContentViews/Publish/CVPublishFinish.js

```diff
@@ -0,0 +1,149 @@
+import { STATUS } from 'foremanReact/constants';
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { useDispatch, useSelector, shallowEqual } from 'react-redux';
+import { Bullseye, Progress, ProgressVariant, Button, Stack, StackItem } from '@patternfly/react-core';
+import { ChartDonut } from '@patternfly/react-charts';
+import { ExternalLinkAltIcon } from '@patternfly/react-icons';
+import { Redirect } from 'react-router-dom';
+import {
+  selectPublishContentViewsError, selectPublishContentViews,
+  selectPublishContentViewStatus,
+} from './ContentViewPublishSelectors';
+import { selectPublishTaskPoll, selectPublishTaskPollStatus } from '../Details/ContentViewDetailSelectors';
+import getContentViews, { publishContentView } from '../ContentViewsActions';
+import getContentViewDetails, { pollPublishTask, handleFinishedTask } from '../Details/ContentViewDetailActions';
+import Loading from '../../../components/Loading';
+import EmptyStateMessage from '../../../components/Table/EmptyStateMessage';
+import { cvVersionPublishKey } from '../ContentViewsConstants';
+
+
+const CVPublishFinish = ({
+  cvId,
+  checkedItems, setCheckedItems,
+  forcePromote, description, setDescription,
+  setIsOpen, versionCount,
+}) => {
+  const dispatch = useDispatch();
+  const [redirect, setRedirect] = useState(false);
+  const [saving, setSaving] = useState(true);
+  const [polling, setPolling] = useState(false);
+  const [taskErrored, setTaskErrored] = useState(false);
+  const [taskErrors, setTaskErrors] = useState(null);
+  const response = useSelector(state => selectPublishContentViews(state, cvId, versionCount));
+  const status = useSelector(state => selectPublishContentViewStatus(state, cvId, versionCount));
+  const error = useSelector(state => selectPublishContentViewsError(state, cvId, versionCount));
+  const pollResponse = useSelector(state =>
+    selectPublishTaskPoll(state, cvVersionPublishKey(cvId, versionCount)), shallowEqual);
+  const pollResponseStatus = useSelector(state =>
+    selectPublishTaskPollStatus(state, cvVersionPublishKey(cvId, versionCount)), shallowEqual);
+
+  useEffect(() => {
+    setSaving(true);
+    dispatch(publishContentView({
+      id: cvId,
+      versionCount,
+      description,
+      environment_ids: checkedItems,
+      is_force_promote: (forcePromote.length > 0),
+    }));
+    setDescription('');
+    setCheckedItems([]);
+  }, []);
+
+  useEffect(() => {
+    setSaving(true);
+    const { id } = response;
+    if (id && status === STATUS.RESOLVED) {
+      setSaving(false);
+      dispatch(pollPublishTask(cvVersionPublishKey(cvId, versionCount), response));
+      setPolling(true);
+    } else if (status === STATUS.ERROR) {
+      setSaving(false);
+    }
+  }, [JSON.stringify(response), status, error]);
```

Comment:
I still hate `JSON.stringify`.. Have you thought any more about adding that `useDeepCompareEffect`?

Reply:
I tried out the deepcompare locally..Seems to work except I was hitting some snags on the filter page for some reason. 
I am adding a card to the board to take care of that on all pages. https://projects.theforeman.org/issues/32621